### PR TITLE
[plugin.video.yelo] 0.0.5

### PR DESCRIPTION
**Pull request by shycoderX**
### Description
This plugin makes it possible to watch live streams that are available on
the Belgium Telenet Yelo application.

All the plugin does is create some simple http requests to accomplish this, alongside with
widevine which handles the DRM content as the live streams are DRM protected.

For the widevine transaction the inputstream.adaptive plugin by peak3d is being used.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.yelo/resources/lib/yelo/yelo.py [hunk 1]
```diff
@@ -171,18 +182,29 @@ def fetch_CallbackKey(self):
...
+        # if postal is not filled in, in settings try to retrieve it from customerFeatures
         if not postal.are_filled_in():
-            self.kodi_wrapper.dialog_ok("Postal", "Please enter your postal code in settings.")
-            self.kodi_wrapper.open_settings()
-            postal.reload()
+            postal.postal_code = self.fetch_Postal()
+
+            # if that does not seem to work..
+            if postal.postal_code is None:
+                self.kodi_wrapper.dialog_ok("Postal", "Please enter your postal code in settings.")
```
Comment (enen92) [inline plugin.video.yelo/resources/lib/yelo/yelo.py]: Can you localize these strings?

File: plugin.video.yelo/resources/lib/yelo/yelo.py [hunk 2]
```diff
@@ -304,9 +319,17 @@ def select_manifest_url(self, channel):
                                       False, None, self.testing)
 
                 j = r.json()
-                return j["stream"]["streamDescriptor"]["manifest"]
+
+                if not j.get("errors"):
+                    return j["stream"]["streamDescriptor"]["manifest"]
+                else:
+                    if j["errors"][0]["code"] == "EPG_RESTRICTED_EUROPE":
+                        self.kodi_wrapper.dialog_ok("RESTRICTED",
```
Comment (enen92) [inline plugin.video.yelo/resources/lib/yelo/yelo.py]: Same

File: plugin.video.yelo/resources/lib/yelo/yelo.py [hunk 3]
```diff
@@ -304,9 +319,17 @@ def select_manifest_url(self, channel):
...
+                                                    "This channel is only available within Europe.")
+                    elif j["errors"][0]["code"] == "EPG_RESTRICTED_NOWHERE":
+                        self.kodi_wrapper.dialog_ok("RESTRICTED",
```
Comment (enen92) [inline plugin.video.yelo/resources/lib/yelo/yelo.py]: Same

File: plugin.video.yelo/resources/lib/yelo/yelo.py [hunk 4]
```diff
@@ -84,15 +84,20 @@ def login(self):
...
-        self._request_OAuthTokens()
-        self._request_entitlement()
+        if self._callback(callback_url):
+            self._verify_token()
+            self._request_OAuthTokens()
+            self._request_entitlement_and_postal()
+
+            return True
+        else:
+            self.kodi_wrapper.dialog_ok("Login failed",
```
Comment (Lunatixz) [inline plugin.video.yelo/resources/lib/yelo/yelo.py]: Can you use localized strings, thx.

**Comment by shycoderX** [comment]
> This add-on isn't Python 3 compatible. As mentioned at https://kodi.tv/article/attention-addon-developers-migration-python-3 only Python 3 compatible add-ons are accepted into the leia branch.

But Leia is still Python 2? If I make it compatible with Python 3 would it still work on Python 2?

**Comment by Rechi** [comment]
Yes it's possible to keep it Python 2 compatible.

**Comment by shycoderX** [comment]
> Yes it's possible to keep it Python 2 compatible.

Done.

File: plugin.video.yelo/resources/lib/yelo/yelo.py (hunk 4 above)
Comment (shycoderX) [reply plugin.video.yelo/resources/lib/yelo/yelo.py]: Done

**Pull request merged**